result/interfaces.go

```diff
@@ -16,5 +16,5 @@ type Result interface {
 // Slave defines the operations that are accessible on a slave
 type Slave interface {
 	// Process performs the work that's dictated by the master
-	Process() error
+	Process() (Result,error)
```

Comment:
I think this should return a pointer to a `Result`: `Process() (*Result, error)`.